### PR TITLE
Secvol

### PR DESCRIPTION
**Pull request by witlox**
added multiple security groups fix
added ability to boot from volume

### CODE REVIEW

**Comment by riccardomurri** [comment]
Please merge the PR I have sent you regarding this one.

File: docs/configure.rst [hunk 1]
```diff
@@ -732,15 +732,21 @@ Optional configuration keys
 
 ``boot_disk_type``
     Define the type of boot disk to use.
-    Only supported when the cloud provider is `google`.
-    Supported values are `pd-standard` and `pd-ssd`.
-    Default value is `pd-standard`.
+    Only supported when the cloud provider is `google` or `openstack`.
+    Supported values are `pd-standard` and `pd-ssd` for Google,
+    or the types available in the openstack volume (cinde) configuration.
```
Comment (riccardomurri) [inline docs/configure.rst]: cinde -> cinder

File: docs/configure.rst [hunk 2]
```diff
@@ -732,15 +732,21 @@ Optional configuration keys
...
+    the boot_disk_size when using this with openstack.
 
 ``boot_disk_size``
     Define the size of boot disk to use.
-    Only supported when the cloud provider is `google`.
+    Only supported when the cloud provider is `google` or `openstack`.
     Values are specified in gigabytes.
-    Default value is 10.
+    Default value for Google is 10.
+    No default is given for openstack.
```
Comment (riccardomurri) [inline docs/configure.rst]: "No default ... for openstack" -> so what happens if I do not give a value?

File: docs/configure.rst [hunk 3]
```diff
@@ -732,15 +732,21 @@ Optional configuration keys
...
 ``boot_disk_type``
     Define the type of boot disk to use.
-    Only supported when the cloud provider is `google`.
-    Supported values are `pd-standard` and `pd-ssd`.
-    Default value is `pd-standard`.
+    Only supported when the cloud provider is `google` or `openstack`.
+    Supported values are `pd-standard` and `pd-ssd` for Google,
+    or the types available in the openstack volume (cinde) configuration.
+    Default value is `pd-standard` for Google.
+    When using this option for openstack, it creates volumes to be used
```
Comment (riccardomurri) [inline docs/configure.rst]: openstack -> OpenStack (here and elsewhere)

**Comment by witlox** [comment]
- no type checking on `boot_disk_size` or `boot_disk_type`: setting defaults invalidates the control flow for the changes, currently there are defaults being set in gce.py, but I cannot think of any descent defaults for OpenStack, for example the name of the volume type is configured per OpenStack instance...

- is there really nothing better than polling, in order to know when a volume is ready? Seeing that the python-cinder client doesn't have any signalling on board, I don't really know of a better solution that won't create a lot of plumbing code.....

**Comment by riccardomurri** [comment]
> > _(RM:)_ no type checking on boot_disk_size or boot_disk_type: 

> _(PW:)_ setting defaults invalidates the control flow for the changes

???

What I mean is: there are lines of code like this:

    bds = int(kwargs['boot_disk_size'])
    
If someone has, e.g., `boot_disk_size=10gb` in the config file, the code above
will trigger a `ValueError` deep down the ElastiCluster call stack in
`OpenStackCloudProvider.start_instance()` which will result in an error message
like `VM "xxx" cannot be started`. A `ConfigurationError` should be the thing
users see instead!

**Comment by riccardomurri** [comment]
Merged into "master" as commit 4373b22 -- thanks a lot for contributing!

**Pull request closed without merging**